### PR TITLE
Automatically register systems with RHSM into Insights

### PR DESCRIPTION
**Pull request by matejmatuska**
If system uses rhsm it is automatically registered into Red Hat Insights
at the end of the upgrade process.
For the registration the `insights-client` package is requiered and will
be installed during the upgrade if missing.

This can be skipped by setting the `LEAPP_NO_INSIGHTS_REGISTER=1`
environment variable or `--no-insights-register` CLI option.

A report is generated informing about the registration and eventual
`insights-client` package installation.

Jira ref.: OAMG-7872

### CODE REVIEW

**Comment by pirat89** [comment]
rebased

**Comment by fernflower** [comment]
Do you have an integration test for that?

File: repos/system_upgrade/common/actors/insightsautoregister/libraries/insightsautoregister.py
Comment (fernflower) [inline repos/system_upgrade/common/actors/insightsautoregister/libraries/insightsautoregister.py:10]: Can this actually produce OSError?

File: repos/system_upgrade/common/actors/checkinsightsautoregister/libraries/checkinsightsautoregister.py
Comment (fernflower) [inline repos/system_upgrade/common/actors/checkinsightsautoregister/libraries/checkinsightsautoregister.py:15]: s/poduced/produced

**Review by fernflower — APPROVED** [review]
LGTM.
IIUC this pr doesn't need to introduce any extra integration tests, just to make sure the existing regression tests are fine. There is some infra issue with copr-build action (not related to the patch), once the tests pass I will approve the pr.

**Review by fernflower — COMMENTED** [review]
Removing approval until infra issue resolved

File: repos/system_upgrade/common/actors/checkinsightsautoregister/libraries/checkinsightsautoregister.py
Comment (matejmatuska) [reply repos/system_upgrade/common/actors/checkinsightsautoregister/libraries/checkinsightsautoregister.py:15]: Interesting, the codespell action should have caught this I think.

File: repos/system_upgrade/common/actors/insightsautoregister/libraries/insightsautoregister.py
Comment (matejmatuska) [reply repos/system_upgrade/common/actors/insightsautoregister/libraries/insightsautoregister.py:10]: The `insights-client` package might fail to install. I think I actually need FileNotFoundError in this case

Comment (matejmatuska) [reply repos/system_upgrade/common/actors/insightsautoregister/libraries/insightsautoregister.py:10]: That should be handled by `run` actually. I will remove the OSError

**Review by fernflower — COMMENTED** [review]
Code looks ok. This needs to be manually tested as our regression tests don't run with rhsm

**Comment by fernflower** [comment]
Works as expected, upgrade is successful, package installed if not present.
Just one NIT - I'd probably add some actor description not to have "There has been no description provided for this actor" in the logs.
Screenshot 2023-02-08 at 12-53-50 noVNC

**Review by fernflower — APPROVED** [review]
I'll just wait for the regression tests for the peace of mind and merge it

**Pull request merged**
